### DISCUSSION TITLE
UE4.27 Build Project: UEReferenceApp failed

### DISCUSSION THREAD

**Discussion by hongzhou91** [JSBSim-Team/jsbsim#703] [Q&A]
Build started...
1>------ Build started: Project: UEReferenceApp, Configuration: Development_Editor x64 ------
1>Creating makefile for UEReferenceAppEditor (no existing makefile)
1>Parsing headers for UEReferenceAppEditor
1>  Running UnrealHeaderTool "D:\jsbsim-master-427\UnrealEngine\UEReferenceApp.uproject" "D:\jsbsim-master-427\UnrealEngine\Intermediate\Build\Win64\UEReferenceAppEditor\Development\UEReferenceAppEditor.uhtmanifest" -LogCmds="loginit warning, logexit warning, logdatabase error" -Unattended -WarningsAsErrors -abslog="C:\Users\AMD001\AppData\Local\UnrealBuildTool\Log_UHT.txt" -installed
1>LogInit: Display: Loading text-based GConfig....
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(29) : LogCompile: Error: Type 'double' is not supported by blueprint. Tank.FuelDensityPoundsPerGallon
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(31) : LogCompile: Error: Type 'double' is not supported by blueprint. Tank.ContentGallons
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(36) : LogCompile: Error: Type 'double' is not supported by blueprint. Tank.CapacityGallons
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(39) : LogCompile: Error: Type 'double' is not supported by blueprint. Tank.FillPercentage
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(42) : LogCompile: Error: Type 'double' is not supported by blueprint. Tank.TemperatureCelcius
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(67) : LogCompile: Error: Type 'double' is not supported by blueprint. Gear.NormalizedPosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(93) : LogCompile: Error: Type 'double' is not supported by blueprint. Gear.WheelRollLinearVelocityMetersPerSec
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(153) : LogCompile: Error: Type 'double' is not supported by blueprint. EngineCommand.Throttle
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(157) : LogCompile: Error: Type 'double' is not supported by blueprint. EngineCommand.Mixture
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(164) : LogCompile: Error: Type 'double' is not supported by blueprint. EngineCommand.PropellerAdvance
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(222) : LogCompile: Error: Type 'double' is not supported by blueprint. EngineState.Thrust
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(224) : LogCompile: Error: Type 'double' is not supported by blueprint. EngineState.EngineRPM
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(227) : LogCompile: Error: Type 'double' is not supported by blueprint. EngineState.N1
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(229) : LogCompile: Error: Type 'double' is not supported by blueprint. EngineState.N2
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(280) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.Aileron
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(283) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.Elevator
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(286) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.Rudder
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(290) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.YawTrim
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(293) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.PitchTrim
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(296) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.RollTrim
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(302) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.Steer
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(305) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.LeftBrake
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(308) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.RightBrake
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(311) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.CenterBrake
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(314) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.ParkingBrake
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(318) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.GearDown
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(324) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.Flap
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(327) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.SpeedBrake
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(330) : LogCompile: Error: Type 'double' is not supported by blueprint. FlightControlCommands.Spoiler
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(358) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.ElevatorPosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(360) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.LeftAileronPosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(362) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.RightAileronPosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(364) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.RudderPosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(366) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.FlapPosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(368) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.SpeedBrakePosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(370) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.SpoilersPosition
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(375) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.CalibratedAirSpeedKts
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(377) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.GroundSpeedKts
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(379) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.TotalVelocityKts
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(383) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.AltitudeASLFt
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(385) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.AltitudeAGLFt
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(387) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.AltitudeRateFtps
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(389) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.StallWarning
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(396) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.Latitude
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/FDMTypes.h(398) : LogCompile: Error: Type 'double' is not supported by blueprint. AircraftState.Longitude
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/JSBSimMovementComponent.h(119) : LogCompile: Error: Type 'double' is not supported by blueprint. JSBSimMovementComponent.FlapPositionAtStart
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/JSBSimMovementComponent.h(125) : LogCompile: Error: Type 'double' is not supported by blueprint. JSBSimMovementComponent.InitialCalibratedAirSpeedKts
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/JSBSimMovementComponent.h(141) : LogCompile: Error: Type 'double' is not supported by blueprint. JSBSimMovementComponent.WindIntensityKts
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/JSBSimMovementComponent.h(154) : LogCompile: Error: Type 'double' is not supported by blueprint. JSBSimMovementComponent.TemperatureCelsius
1>D:/jsbsim-master-427/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Public/JSBSimMovementComponent.h(160) : LogCompile: Error: Type 'double' is not supported by blueprint. JSBSimMovementComponent.PressureSeaLevelhPa
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.MakeFile.Targets(45,5): error MSB3073: The command ""D:\Program Files\Epic Games\UE_4.27\Engine\Build\BatchFiles\Build.bat" UEReferenceAppEditor Win64 Development -Project="D:\jsbsim-master-427\UnrealEngine\UEReferenceApp.uproject" -WaitMutex -FromMsBuild" exited with code 6.
1>Done building project "UEReferenceApp.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

It semm not support UE4.27?

**Comment by gallonmate** [comment]
Correct, right now only UE5 is supported by the plugin.
